fbw-common/src/wasm/systems/systems/src/communications/amu.rs

```diff
@@ -0,0 +1,861 @@
+        );
+        writer.write(
+            &self.receive_com2_id,
+            self.mixed_audio.receive_com2 && self.vhfs[1].is_powered(),
+        );
+        writer.write(
+            &self.receive_adf1_id,
+            self.mixed_audio.receive_adf1 && self.adfs[0].is_powered(),
+        );
+        writer.write(
+            &self.receive_adf2_id,
+            self.mixed_audio.receive_adf2 && self.adfs[1].is_powered(),
+        );
+        writer.write(
+            &self.receive_vor1_id,
+            self.mixed_audio.receive_vor1 && self.vors[0].is_powered(),
+        );
+        writer.write(
+            &self.receive_vor2_id,
+            self.mixed_audio.receive_vor2 && self.vors[1].is_powered(),
+        );
+        writer.write(
+            &self.receive_ils_id,
+            self.mixed_audio.receive_ils && self.ils.is_powered(),
+        );
+
+        // FOR FUTURE USE: Not needed for the time being as there's no K event for all this
+        // writer.write(&self.receive_gls_id, self.mixed_audio.receive_gls);
+        // writer.write(&self.receive_hf1_id, self.receive_hf1);
+        // writer.write(&self.receive_hf2_id, self.receive_hf2);
+        // writer.write(&self.receive_mech_id, self.receive_mech);
+        // writer.write(&self.receive_att_id, self.receive_att);
+        // writer.write(&self.receive_pa_id, self.receive_pa);
+
+        // Special case for markers as there's no XXXX_SET function. Only Toggle.
+        if self.mixed_audio.receive_markers && !self.mixed_audio.sound_markers
+            || !self.mixed_audio.receive_markers && self.mixed_audio.sound_markers
+        {
+            writer.write(&self.receive_markers_id, !self.mixed_audio.sound_markers);
+        }
+
+        writer.write(&self.volume_com1_id, self.mixed_audio.volume_com1);
+        writer.write(&self.volume_com2_id, self.mixed_audio.volume_com2);
+        writer.write(&self.volume_com3_id, self.mixed_audio.volume_com3);
+        writer.write(&self.volume_adf1_id, self.mixed_audio.volume_adf1);
+        writer.write(&self.volume_adf2_id, self.mixed_audio.volume_adf2);
+        writer.write(&self.volume_vor1_id, self.mixed_audio.volume_vor1);
+        writer.write(&self.volume_vor2_id, self.mixed_audio.volume_vor2);
+        writer.write(&self.volume_ils_id, self.mixed_audio.volume_ils);
+
+        // FOR FUTURE USE: Not needed for the time being as there's no K event for all this
+        // writer.write(&self.volume_markers_id, self.mixed_audio.volume_markers);
+        // writer.write(&self.volume_gls_id, self.mixed_audio.volume_gls);
+        // writer.write(&self.volume_hf1_id, self.volume_hf1);
+        // writer.write(&self.volume_hf2_id, self.volume_hf2);
+        // writer.write(&self.volume_att_id, self.volume_att);
+        // writer.write(&self.volume_mech_id, self.volume_mech);
+        // writer.write(&self.volume_pa_id, self.volume_pa);
+    }
+}
+
+pub struct Computer {
```

Comment:
```suggestion
struct Computer {
```
And also remove `pub` from all the functions of this struct.